packages/nextra-theme-docs/src/index.tsx

```diff
@@ -188,11 +177,21 @@ const InnerLayout = ({
       </div>
     )
 
+  const { locale = DEFAULT_LOCALE } = useRouter()
+  const localeConfig = config.i18n.find(l => l.locale === locale)
+  const isRTL = localeConfig
+    ? localeConfig.direction === 'rtl'
+    : config.direction === 'rtl'
+
+  const direction = isRTL ? 'rtl' : 'ltr'
+
   return (
-    <div
-      dir={isRTL ? 'rtl' : 'ltr'}
-      className={cn('nextra-container transition-opacity', menu && 'opacity-80')}
-    >
+    <>
+      <script
+        dangerouslySetInnerHTML={{
+          __html: `document.documentElement.setAttribute('dir','${direction}')`
```

Comment:
Makes sense! Does it work when the language changes like from LTR to RTL?

Reply:
yeah it works, also there is no shrinking on initial loading